StackOverflow/Controller/QuestionsTableViewController.m

```diff
@@ -0,0 +1,97 @@
+//
+//  QuestionsTableViewController.m
+//  StackOverflow
+//
+//  Created by Mali on 2017/10/04.
+//  Copyright © 2017 DVT. All rights reserved.
+//
+
+#import "QuestionsTableViewController.h"
+#import "QuestionsCell.h"
+#import "DataFetchDelegate.h"
+#import "QuestionsDataObject.h"
+#import "QuestionsData.h"
+
+
+@interface QuestionsTableViewController () <DataFetchDelegate> {
+    NSArray *dataResult;
+    QuestionsDataObject *dataObject;
+    UIActivityIndicatorView *activityIndicator;
+}
+
+@end
+
+@implementation QuestionsTableViewController
+
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    //Display a code loading indicator
+    activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+    activityIndicator.alpha = 1.0;
+    [self.view addSubview:activityIndicator];
+    activityIndicator.center = CGPointMake([[UIScreen mainScreen]bounds].size.width/2, [[UIScreen mainScreen]bounds].size.height/2);
+    [activityIndicator startAnimating];
+
+    dataObject = [[QuestionsDataObject alloc] init];
+    dataObject.questionRequest = [[QuestionRequest alloc] init];
+    dataObject.questionRequest.delegate = dataObject;
+    dataObject.delegate = self;
+    
+    [dataObject getData];
+    
+}
+
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return dataResult.count;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    QuestionsCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
+    
+    QuestionsData *questionData = dataResult[indexPath.row];
+    
+    //Select image based on whether the answer is accepted or not
```

Comment:
The configuration of the presentation performed by the QuestionCell should not be done inside the DataSource for several reasons. Firstly, the state and layout of the QuestionCell is private to that cell and therefore you violate encapsulation ( a primary tenet of OO ) by configuring the cell inside the DataSource implementation. Secondly if you ever reuse the QuestionCell in another controller, you will have to recode all the configuration logic in that DataSource as well. Define a setter on the QuestionCell that accepts the question, save the model ( i.e. the question ) in the cell, and then in that setter call another method on the QuestionCell that actually configures the presentation from the stored question. If you need me to explain this please contact me.